src/functionalTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringFunctionalTest.java

```diff
@@ -59,6 +59,29 @@ void shouldRetrieveCourtVenues_By_SearchString_WithStatusCode_200() {
         assertThat(courtNameVerified.size()).isPositive();
     }
 
+    @Test
+    @ToggleEnable(mapKey = mapKey, withFeature = true)
+    void shouldRetrieveCourtVenues_By_SearchStringWithHyphen_WithStatusCode_200() {
+        final var response = (LrdCourtVenueResponse[]) lrdApiClient.retrieveResponseForGivenRequest(HttpStatus.OK,
+                          "?search-string=Stoke-on", LrdCourtVenueResponse[].class, path);
+        assertThat(response).isNotEmpty();
+
+        var courtVenueResponse = new ArrayList<>(Arrays.asList(response));
+        var courtNameVerified = courtVenueResponse
+            .stream()
+            .filter(venue -> venue.getCourtName().strip().toLowerCase().contains("Stoke-".toLowerCase())
+                || venue.getSiteName().strip().toLowerCase().contains("Stoke-".toLowerCase())
+                || venue.getCourtAddress().strip().toLowerCase().contains("Stoke-".toLowerCase())
```

Comment:
why these filters are required inside test method ?

Reply:
@rajkatla-hmcts it is to check if each of the court venues returned in the response have one of those fields matching with the search string given. Currently when querying the db we look for the search string in these fields as well: `siteName`, `courtName`, `postcode` and `courtAddress` as you can see here https://github.com/hmcts/rd-location-ref-api/blob/07162e08b238f865302113c884f1c44ba929ddd7/src/main/java/uk/gov/hmcts/reform/lrdapi/repository/CourtVenueRepository.java#L50-L53